libcrux_crypto/src/crypto.rs

```diff
@@ -0,0 +1,578 @@
+use std::sync::{Mutex, MutexGuard};
+
+use libcrux::drbg::{Drbg, RngCore};
+use libcrux::hpke::{self, HPKEConfig};
+use openmls_traits::crypto::OpenMlsCrypto;
+use openmls_traits::types::{
+    AeadType, Ciphersuite, CryptoError, ExporterSecret, HashType, HpkeAeadType, HpkeCiphertext,
+    HpkeConfig, HpkeKdfType, HpkeKemType, HpkeKeyPair, KemOutput, SignatureScheme,
+};
+
+use rand::CryptoRng;
+use tls_codec::SecretVLBytes;
+
+const MAX_DATA_LEN: usize = 0x10000000;
+
+/// The libcrux-backed cryptography provider for OpenMLS
+pub struct CryptoProvider {
+    drbg: Mutex<Drbg>,
+}
+
+impl Default for CryptoProvider {
+    fn default() -> Self {
```

Comment:
Is it that important for the provider to implement `Default` given that there are two unwraps here?

Reply:
Don't we use default all over the tests in the OpenMLS tests? If we don't implement it, I don't think we would be able to run the tests against this provider.

Reply:
Fair enough, in which case the unwraps are okay, as long as the implementation is qualified with the right config flags.